scripts/tests/twister_blackbox/test_coverage.py

```diff
@@ -0,0 +1,294 @@
+                'coverage'
+            ],
+            r'{"files": \[], "gcovr/format_version": ".*"}'
+        ),
+    ]
+    TESTDATA_4 = [
+        (
+            'gcovr',
+            [
+                'coverage.log', 'coverage.json',
+                'coverage', os.path.join('coverage','coverage.xml')
+            ],
+            'xml'
+        ),
+        (
+            'gcovr',
+            [
+                'coverage.log', 'coverage.json',
+                'coverage', os.path.join('coverage','coverage.sonarqube.xml')
+            ],
+            'sonarqube'
+        ),
+        (
+            'gcovr',
+            [
+                'coverage.log', 'coverage.json',
+                'coverage', os.path.join('coverage','coverage.txt')
+            ],
+            'txt'
+        ),
+        (
+            'gcovr',
+            [
+                'coverage.log', 'coverage.json',
+                'coverage', os.path.join('coverage','coverage.csv')
+            ],
+            'csv'
+        ),
+        (
+            'gcovr',
+            [
+                'coverage.log', 'coverage.json',
+                'coverage', os.path.join('coverage','coverage.coveralls.json')
+            ],
+            'coveralls'
+        ),
+        (
+            'gcovr',
+            [
+                'coverage.log', 'coverage.json',
+                'coverage', os.path.join('coverage','index.html')
+            ],
+            'html'
+        ),
+        (
+            'lcov',
+            [
+                'coverage.log', 'coverage.info',
+                'ztest.info', 'coverage',
+                os.path.join('coverage','index.html')
+            ],
+            'html'
+        ),
+        (
+            'lcov',
+            [
+                'coverage.log', 'coverage.info',
+                'ztest.info'
+            ],
+            'lcov'
+        ),
+    ]
+    TESTDATA_5 = [
+        (
+            os.path.join(TEST_DATA, 'tests', 'dummy', 'agnostic', 'group2'),
+            ['qemu_x86'],
+            'gcovr',
+            'Running gcovr -r'
+        ),
+        (
+            os.path.join(TEST_DATA, 'tests', 'dummy', 'agnostic', 'group2'),
+            ['qemu_x86'],
+            'lcov',
+            'Running lcov --gcov-tool'
+        )
+    ]
+
+    @classmethod
+    def setup_class(cls):
+        apath = os.path.join(ZEPHYR_BASE, 'scripts', 'twister')
+        cls.loader = importlib.machinery.SourceFileLoader('__main__', apath)
+        cls.spec = importlib.util.spec_from_loader(cls.loader.name, cls.loader)
+        cls.twister_module = importlib.util.module_from_spec(cls.spec)
+
+    @pytest.mark.parametrize(
+        'test_path, test_platforms, file_name',
+        TESTDATA_1,
+        ids=[
+            'coverage',
+        ]
+    )
+    def test_coverage(self, capfd, test_path, test_platforms, out_path, file_name):
+        args = ['-i','--outdir', out_path, '-T', test_path, '--coverage', '--coverage-tool', 'gcovr'] + \
```

Comment:
I like to split args into separate lists, to have a clear picture of what are the flags under tests.

Reply:
done, now it's more clear